apps/web/src/services/dashboards/shared/dashboard-widget-input-form/dashboard-widget-options-form/DashboardCostWidgetValueOptionDropdown.vue

```diff
@@ -1,55 +1,203 @@
 <script setup lang="ts">
-import { PIconButton, PSelectDropdown } from '@spaceone/design-system';
+import {
+    reactive, watch,
+} from 'vue';
+
+import { PIconButton, PSelectDropdown, PBadge } from '@spaceone/design-system';
 import type {
     AutocompleteHandler,
     SelectDropdownMenuItem,
 } from '@spaceone/design-system/types/inputs/dropdown/select-dropdown/type';
 
+import { SpaceConnector } from '@cloudforet/core-lib/space-connector';
+
+import ErrorHandler from '@/common/composables/error/errorHandler';
+
+import { useWidgetFormStore } from '@/services/dashboards/shared/dashboard-widget-input-form/widget-form-store';
+
 interface Props {
+    optionKey?: string;
     selected?: SelectDropdownMenuItem[];
-    menuHandlers?: AutocompleteHandler[];
     invalid?: boolean;
 }
 
 const props = withDefaults(defineProps<Props>(), {
+    optionKey: '',
     selected: undefined,
-    menuHandlers: undefined,
     invalid: false,
 });
 
 const emit = defineEmits<{(e: 'update:selected', selected: SelectDropdownMenuItem[]): void;
     (e: 'delete'): void;
 }>();
 
-const handleUpdateSelected = (selected: SelectDropdownMenuItem[]) => {
-    emit('update:selected', selected);
+const widgetFormStore = useWidgetFormStore();
+const widgetFormState = widgetFormStore.$state;
+
+const state = reactive({
+    visibleMenu: false,
+    reloadOnMenuHandlerUpdate: false,
+    selectedKey: {} as SelectDropdownMenuItem,
+    valueTarget: props.optionKey === 'cost_tag_value' ? 'tags' : 'additional_info',
+    selectedItems: [] as SelectDropdownMenuItem[],
+});
+
+const variableModelKeys = {
+    cost_tag_value: 'cost_tag_keys',
+    cost_additional_info_value: 'cost_additional_info_keys',
+} as const;
+
+const menuHandler: AutocompleteHandler = async () => {
+    // key handler
+    if (Object.keys(state.selectedKey).length === 0) {
+        const param = {
+            data_source_id: widgetFormState.widgetOptions.cost_data_source,
+            query: {
+                only: [variableModelKeys[props.optionKey]],
+                filter: [
+                    {
+                        key: 'cost_tag_keys',
+                        value: null,
+                        operator: 'not',
+                    },
+                ],
+            },
+        };
+        const { results } = await SpaceConnector.clientV2.costAnalysis.dataSource.list(param);
+        return {
+            results: results[0][variableModelKeys[props.optionKey]].map((result) => (
+                { label: result, name: `${state.valueTarget}.${result}` }
+            )),
+        };
+    }
+    // value handler
+    const param = {
+        resource_type: 'cost_analysis.Cost',
+        options: { limit: 10, search_type: 'value' },
+        distinct_key: state.selectedKey.name,
+    };
+    try {
+        const res = await SpaceConnector.client.addOns.autocomplete.distinct(param);
```

Comment:
issue: this api is deprecated. Please change to use the same api(SpaceConnector.clientV2.costAnalysis.dataSource.list).

Reply:
The API path has been updated to 'SpaceConnector.clientV2.costAnalysis.cost.stat'.
Please make a note of this change!